dask/tests/test_order.py

```diff
@@ -2411,6 +2413,78 @@ def test_connecting_to_roots_tree_reduction():
     assert all(v == 2 for v in max_dependents.values())
 
 
+def test_connecting_to_roots_long_linear_chains():
+    dsk = {
+        "a0": (f, 1),
+        "a1": (f, 1),
+        "b0": (f, "a0"),
+        "c0": (f, "b0"),
+        "d0": (f, "c0"),
+        "d1": (f, "c0"),
+        "e1": (f, "d0", "d1"),
+        "b1": (f, "a1"),
+        "c1": (f, "b1"),
+        "d2": (f, "c1"),
+        "d3": (f, "c1"),
+        "f": (f, "e1", "d2", "d3", "d1"),
+    }
+    dependencies, dependents = get_deps(dsk)
+    connected_roots, _ = _connecting_to_roots(dependencies, dependents)
+
+    id_a0 = id(connected_roots["b0"])
+    for k in ["c0", "d0", "d1", "e1"]:
+        assert id(connected_roots[k]) == id_a0
+
+    id_a0 = id(connected_roots["b1"])
+    for k in ["c1", "d2", "d3"]:
+        assert id(connected_roots[k]) == id_a0
+
+
+def test_connected_roots_deduplication_mem_usage():
+    # see https://github.com/dask/dask/issues/11055
+    np = pytest.importorskip("numpy")
+    da = pytest.importorskip("dask.array")
+
+    def rotate(X, n_iter=2):
+        gamma = 1
+        n_samples, n_modes = X.shape
+        R = np.eye(n_modes)
+
+        for _ in range(n_iter):
+            basis = X @ R
+
+            basis2 = basis * basis.conj()
+            basis3 = basis2 * basis
+            W = np.diag(np.sum(basis2, axis=0))
+            alpha = gamma / n_samples
+
+            transformed = X.conj().T @ (basis3 - (alpha * basis @ W))
+            U, svals, VT = da.linalg.svd_compressed(
+                transformed, k=n_modes, compute=False
+            )
+            R = U @ VT
+
+        Xrot = X @ R
+        return Xrot
+
+    n_modes = 100
+
+    x = da.random.random((10_000, 100), chunks=(5_000, -1))
```

Comment:
This deviation of the example from the issue is intended?

Reply:
Yes, the example reported in the issue is really big and this is just a smaller version that exhibits all the interesting things.